cpp/state/c_state.h

```diff
@@ -17,6 +17,11 @@
 extern "C" {
 #endif
 
+// Macro to duplicate function declarations for Rust and CPP
+#define DUPLICATE_FOR_LANGS(ret_type, fn) \
+  ret_type Rust_##fn;                     \
+  ret_type CPP_##fn;
```

Comment:
Structurally, "Carmen_Rust_XXX" and "Carmen_Cpp_XXX" would be more suiting. You could include the "Carmen_" prefix here and remove it from the functions - no need to try to past that in half-way through the function name.

Wdty?